### PR TITLE
Feature/edit transactions

### PR DESCRIPTION
**Pull request by annaleaschroeder**
- added edit-transaction logic, created new "Edit-Transaction-Page" 
- rewrote parts of form component to enable re-initialization with existing transaction-data 
- outsourced "Return to overview"-Button to new component

For further details, refer to this [issue](https://github.com/annaleaschroeder/capstone-project/issues/27)

### CODE REVIEW

File: src/components/ReturnToOverviewButton.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+`
+
+const ReturnRoute = styled.a`
+  text-decoration: none;
+`
+
+const ReturnButton = styled.button`
+  display: block;
+  border: none;
+  background-color: white;
```
Comment (LebkuchenCJ) [inline src/components/ReturnToOverviewButton.js]: Why not working with CSS variables as you did in the other styling? Also, I found white to be hard to differentiate from the white background. 
image

File: src/components/TransactionFormInput.js [hunk 2]
```diff
@@ -9,64 +9,77 @@ TransactionFormInput.propTypes = {
...
-            <option value="Miscellaneous">Miscellaneous</option>
-            <option value="Earnings">Earnings</option>
-          </DropDown>
+      {({ values, errors, touched }) => {
+        return (
+          <FormStyled>
+            <Input
+              name="value"
+              placeholder="Enter transaction."
+              value={values.value}
```
Comment (LebkuchenCJ) [inline src/components/TransactionFormInput.js:46]: I find the naming of your cariables here a little bit confusing. Maybe you be more specific, what kind of value this is?

File: src/components/TransactionFormInput.js [hunk 3]
```diff
@@ -9,64 +9,77 @@ TransactionFormInput.propTypes = {
...
+              <ErrorDropDownStyled>{errors.tag}</ErrorDropDownStyled>
+            ) : null}
+            <Notes
+              type="textarea"
+              name="notes"
+              placeholder="Add notes to transaction"
+              value={values.notes}
+            />
+            {errors.notes && touched.notes ? (
+              <ErrorMessageNotesStyled>{errors.notes}</ErrorMessageNotesStyled>
```
Comment (LebkuchenCJ) [inline src/components/TransactionFormInput.js]: Could this be possible, too? Styled seems a little bit too much here
```suggestion
              <ErrorMessageNotes>{errors.notes}</ErrorMessageNotes>
```

File: src/components/pages/EditTransactionPage.js [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+  function handleToggle() {
+    setSelected(!selected)
+  }
+
+  function onEditUpdateTransactionEntry(value, notes, tag) {
+    const updatedTransaction = {
+      type: selected ? 'income' : 'spending',
+      value: parseFloat(String(value).replace(',', '.')),
+      notes: notes,
+      tag: tag,
```
Comment (LebkuchenCJ) [inline src/components/pages/EditTransactionPage.js:43]: I think when you don't change the object key you can also write them like this 
```suggestion
      notes,
      tag,
```

**Review by LebkuchenCJ — COMMENTED** [review]
hey, nice progress 👍 
only some small things I noticed. And I think this PR could easily be divided into two. This could make reviewing it a tiny bit easier to follow. 
Regarding the styling of your button: I think you could reduce the box shadow a bit as it seems too wide for me. Also you could think of removing the text "RETURN" as the arrow back signalizes to the user to go back  already. (at least for me)
You could think of adding `cursor: pointer` on your button to create smooth experience on window screen. 

One last thing: 
Don't forget to assign yourself and the project here in order to display it to your Kanban board
image

File: src/components/TransactionFormInput.js (hunk 3 above)
Comment (annaleaschroeder) [reply src/components/TransactionFormInput.js]: Good point

File: src/components/pages/EditTransactionPage.js (hunk 4 above)
Comment (annaleaschroeder) [reply src/components/pages/EditTransactionPage.js:43]: Good point. In this case, I decided to leave them in to clarify that there is data that the user can update, and data that will remain unchanged.

File: src/components/TransactionFormInput.js (hunk 2 above)
Comment (annaleaschroeder) [reply src/components/TransactionFormInput.js:46]: Unfortunately, this is a built in functionality in Formik. It ensures that you can actually update the imported values

**Pull request merged**